nixos/modules/services/misc/klipper.nix

```diff
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.klipper;
+  package = pkgs.klipper;
+in
+{
+  ##### interface
+  options = {
+    services.klipper = {
+      enable = mkEnableOption "Klipper, the 3D printer firmware";
+
+      user = mkOption {
+        type = types.str;
+        default = "klipper";
+        description = "User for the daemon.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "klipper";
+        description = "Group for the daemon.";
+      };
+
+      logDir = mkOption {
+        type = types.path;
+        default = "/var/log/klipper";
+        description = "Log directory of the daemon.";
+      };
+
+      extraConfig = mkOption {
```

Comment:
I don't know if this is possible yet, but maybe have a look at https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#an-example for replacing this with `settings`.

Reply:
I tried to convert it to this format. Please check :)